### PR TITLE
Feature/92506 Should be able to add scope across systems

### PR DESCRIPTION
**Pull request by Prestholm**
# Description

Should be able to add scope across systems. This means do not disable scope that is not on the same system as selected scope

Completes: [AB#92506](https://statoildeveloper.visualstudio.com/b6d97460-2e53-47ae-aad4-b41396d90828/_workitems/edit/92506)

# Checklist:

- [X] I have performed a self-review of my own code.
- [X] I have linked my DevOps task using the AB# tag.
- [X] My code is easy to read, and comments are added where needed.
- [ ] My code is covered by tests.

### CODE REVIEW

File: src/modules/InvitationForPunchOut/views/CreateAndEditIPO/SelectScope/CommPkgTable.tsx [hunk 1]
```diff
@@ -79,7 +69,7 @@ const CommPkgTable = forwardRef(
         const tableRef = useRef<any>();
 
         const hasValidSystem = useCallback(
-            (systemString: string): boolean => {
+            (systemString: string) => {
```
Comment (bratli) [inline src/modules/InvitationForPunchOut/views/CreateAndEditIPO/SelectScope/CommPkgTable.tsx:72]: Does this prevent being able to add scope accross sections? The system-attribute looks like this: "system|subsystem" for most plants, but for Heimdal it looks like this "section|system|subsystem".

File: src/modules/InvitationForPunchOut/views/CreateAndEditIPO/SelectScope/CommPkgTable.tsx [hunk 2]
```diff
@@ -188,21 +195,26 @@ const CommPkgTable = forwardRef(
...
             if (commPkgNo == null) {
-                getCommPkgsByFilter(query.pageSize, query.pageIndex).then(
-                    (result) => {
-                        setFilteredCommPkgs(result.commPkgs);
-                        setSelectAll(
-                            result.commPkgs.every((commpkg) =>
-                                hasSameSystem(
-                                    commpkg.system,
-                                    result.commPkgs[0].system
+                if (!filter.trim()) {
```
Comment (bratli) [inline src/modules/InvitationForPunchOut/views/CreateAndEditIPO/SelectScope/CommPkgTable.tsx:198]: Isn't this somehting that was removed in another PR?

**Review by bratli — CHANGES_REQUESTED** [review]

File: src/modules/InvitationForPunchOut/views/CreateAndEditIPO/SelectScope/CommPkgTable.tsx (hunk 2 above)
Comment (Prestholm) [reply src/modules/InvitationForPunchOut/views/CreateAndEditIPO/SelectScope/CommPkgTable.tsx]: Correct, removed now!

File: src/modules/InvitationForPunchOut/views/CreateAndEditIPO/SelectScope/CommPkgTable.tsx [hunk 3]
```diff
@@ -192,12 +199,20 @@ const CommPkgTable = forwardRef(
...
-                            result.commPkgs.every((commpkg) =>
-                                hasSameSystem(
+                            result.commPkgs.every((commpkg) => {
+                                const x = hasSameSection(
                                     commpkg.system,
                                     result.commPkgs[0].system
-                                )
-                            )
+                                );
+                                let y = true;
```
Comment (VRiveland) [inline src/modules/InvitationForPunchOut/views/CreateAndEditIPO/SelectScope/CommPkgTable.tsx:207]: Needs better variable names

**Review by VRiveland — COMMENTED** [review]
I'll fix

**Pull request merged**